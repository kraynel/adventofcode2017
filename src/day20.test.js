const { getClosestToZero, getParticlesAfterCollision } = require("./day20");
const fs = require("fs");

describe("getClosestToZero", () => {
  it("should getClosestToZero", () => {
    expect(
      getClosestToZero(`p=<3,0,0>, v=<2,0,0>, a=<-1,0,0>
p=<4,0,0>, v=<0,0,0>, a=<-2,0,0>`)
    ).toBe(0);

    expect(
      getClosestToZero(`p=<4,0,0>, v=<0,0,0>, a=<-2,0,0>
p=<3,0,0>, v=<2,0,0>, a=<-1,0,0>`)
    ).toBe(1);
  });

  it("should get the value", done => {
    fs.readFile("inputs/day20.txt", "utf8", (err, data) => {
      expect(getClosestToZero(data)).toMatchSnapshot();
      done();
    });
  });
});

describe("getParticlesAfterCollision", () => {
  it("should getParticlesAfterCollision", () => {
    expect(
      getParticlesAfterCollision(`p=<-6,0,0>, v=<3,0,0>, a=<0,0,0>
p=<-4,0,0>, v=<2,0,0>, a=<0,0,0>
p=<-2,0,0>, v=<1,0,0>, a=<0,0,0>
p=<3,0,0>, v=<-1,0,0>, a=<0,0,0>`)
    ).toBe(1);
  });

  it("should get the value", done => {
    fs.readFile("inputs/day20.txt", "utf8", (err, data) => {
      expect(getParticlesAfterCollision(data)).toMatchSnapshot();
      done();
    });
  });
});
