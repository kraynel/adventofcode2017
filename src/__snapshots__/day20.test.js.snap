// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getClosestToZero should get the value 1`] = `457`;

exports[`getParticlesAfterCollision should get the value 1`] = `448`;
